projects/plugins/jetpack/modules/widgets/milestone/class-milestone-widget.php

```diff
@@ -49,6 +49,7 @@ public function __construct() {
 		);
 
 		add_action( 'wp_enqueue_scripts', array( __class__, 'enqueue_template' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
```

Comment:
I would recommend against doing this, as this will mean enqueuing the stylesheet on all pages of a site as soon as the Widgets module is active, even if they don't use that specific widget at all.

One thing you can do instead is only enqueuing the stylesheet if the widget is active, like it's done for this widget for example:
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/modules/widgets/authors.php#L32-L34

Reply:
While working on the `eu_cookie_law_widget` I had success putting the enqueue in a function and calling it in the `widget()` function. Not sure if the same would apply here though 🤷‍♂️

Reply:
Thank you for your feedback Jeremy & Tony.

When I try to enqueue the styles conditionally in the constructor: 

```php
if ( is_active_widget( false, false, $this->id_base, true ) || is_active_widget( false, false, 'monster', true ) || is_customize_preview() ) {
	add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
}
```
or when I try to load them (with the same condition) in the `widget()` function, these styles are not loaded in the Legacy Widget block `iframe` unfortunately.

![Markup on 2021-12-03 at 15:03:32](https://user-images.githubusercontent.com/25105483/144615316-acb5de5a-ac54-4ad4-bfed-83e21f6f0b7b.png)

It is not clear to me how this could be resolved yet. 🤔

Reply:
>  these styles are not loaded in the Legacy Widget block iframe unfortunately.

This may be because `is_customize_preview()` doesn't cover all our possible use-cases here?

Would something like this work better?

https://github.com/Automattic/jetpack/blob/81ea36abcd016aefb9d062f047244340b5a1a100/projects/plugins/jetpack/modules/widgets/contact-info.php#L43-L48

Reply:
It doesn't work either unfortunately.  :/

I tried a couple of things and looked at [how the iframe preview is built in core/Gutenberg](https://github.com/WordPress/gutenberg/blob/d7d2484611fc15e407e1cf2b786b76799320c8b4/packages/widgets/src/blocks/legacy-widget/index.php#L98-L152), but no progress so far.

I will get back to this next week. Maybe it would be worth to build a new Milestone block, hehe. :)

Reply:
Alright. I did a couple of tests and:

When the CSS script is enqueued in the `widget()` - as can be seen in the latest commit (https://github.com/Automattic/jetpack/pull/21904/commits/19060f42f07a3550ce1a6c0aeb37a93353919c28), the CSS is enqueued only when the widget is added to the site. This is true even when we don't enclose the enqueue call in a condition.

So circling back to the original @arcangelini's suggestion above. :)

Similar usage can be found in the `gravatar-profile` widget:

https://github.com/Automattic/jetpack/blob/135bb0b766e4410b1893feedc7f14ddbe94a86a7/projects/plugins/jetpack/modules/widgets/gravatar-profile.php#L97

---

In other words, everything seems to be working as expected now.

Reply:
Just want to say thanks all for chiming in and sharing knowledge on enqueueing stylesheets. :)